components/esp_lvgl_port/esp_lvgl_port.c

```diff
@@ -784,6 +916,323 @@ static void lvgl_port_btn_up_handler(void *arg, void *arg2)
+                    char key = 0;
+
+                    /* LVGL special keys */
+                    if (keyboard->key[i] == HID_KEY_TAB) {
+                        if ((keyboard->modifier.left_shift || keyboard->modifier.right_shift)) {
+                            key = LV_KEY_PREV;
+                        } else {
+                            key = LV_KEY_NEXT;
+                        }
+                    } else if (keyboard->key[i] == HID_KEY_RIGHT) {
+                        key = LV_KEY_RIGHT;
+                    } else if (keyboard->key[i] == HID_KEY_LEFT) {
+                        key = LV_KEY_LEFT;
+                    } else if (keyboard->key[i] == HID_KEY_DOWN) {
+                        key = LV_KEY_DOWN;
+                    } else if (keyboard->key[i] == HID_KEY_UP) {
+                        key = LV_KEY_UP;
+                    } else if (keyboard->key[i] == HID_KEY_ENTER || keyboard->key[i] == HID_KEY_KEYPAD_ENTER) {
+                        key = LV_KEY_ENTER;
+                    } else if (keyboard->key[i] == HID_KEY_DELETE) {
+                        key = LV_KEY_DEL;
+                    } else if (keyboard->key[i] == HID_KEY_HOME) {
+                        key = LV_KEY_HOME;
+                    } else if (keyboard->key[i] == HID_KEY_END) {
+                        key = LV_KEY_END;
+                    } else {
+                        /* Get ASCII char */
+                        key = usb_hid_get_keyboard_char(keyboard->key[i], (keyboard->modifier.left_shift || keyboard->modifier.right_shift));
+                    }
+
+                    if (key == 0) {
+                        ESP_LOGI(TAG, "Not recognized key: %c (%d)", keyboard->key[i], keyboard->key[i]);
+                    }
+                    hid_ctx->kb.last_key = key;
+                    hid_ctx->kb.pressed = true;
+                }
+            }
+
+        } else if (dev.proto == HID_PROTOCOL_MOUSE) {
+            hid_mouse_input_report_boot_t *mouse = (hid_mouse_input_report_boot_t *)data;
+            if (data_length < sizeof(hid_mouse_input_report_boot_t)) {
+                break;
+            }
+            hid_ctx->mouse.left_button = mouse->buttons.button1;
+            hid_ctx->mouse.x += mouse->x_displacement;
+            hid_ctx->mouse.y += mouse->y_displacement;
+        }
+        break;
+    case HID_HOST_INTERFACE_EVENT_TRANSFER_ERROR:
+        break;
+    case HID_HOST_INTERFACE_EVENT_DISCONNECTED:
+        hid_host_device_close(hid_device_handle);
+        break;
+    default:
+        break;
+    }
+}
+
+static void lvgl_port_usb_hid_task(void *arg)
+{
+    hid_host_dev_params_t dev;
+    lvgl_port_usb_hid_ctx_t *ctx = (lvgl_port_usb_hid_ctx_t *)arg;
+    hid_host_device_handle_t hid_device_handle = NULL;
+    vgl_port_usb_hid_event_t msg;
+
+    assert(ctx);
+
+    ctx->running = true;
+
+    while (ctx->running) {
+        if (xQueueReceive(ctx->queue, &msg, pdMS_TO_TICKS(50))) {
```

Comment:
:thinking: 
Yes, it seems that this is a good place to call both of them: 

    while (ctx->running) {
        hid_host_handle_events(pdMS_TO_TICKS(50));
        if (xQueueReceive(ctx->queue, &msg, pdMS_TO_TICKS(50))) { 
            ... 
        }
    }

Reply:
Could we avoid waking up the system every 50ms and block on an event indefinitely? (Using queue sets, `select` or a similar approach.)
Otherwise it will be pretty hard to make this implementation compatible with power management features such as tickless idle and automatic light sleep.

Reply:
@igrr When I put `hid_host_handle_events` into this task, the `xQueueReceive` cannot be blocked indefinitely.

Reply:
I am thinking about the tasks for USB HID and we can make new component, asi @igrr said, it can be something similar like `esp_lvgl_port` for USB HID. But I think, that the best will be put tasks inside the HID component, instead of creating new component. What do you think?

Reply:
Personally, agreed. 
I think that we do not need a separate task at all here. The usage of event_loop can solve this problem and provide the callback which can be used easily (the same way, how the `BT` and `BLE` transport levels are work in `esp_hid`). By the way, this types of optimization (and others, which helps to save some memory) are planned and in process right now. 
As well as possibility to add layouts for a keyboards, 
 
So, the other question, is the LVGL component can be used not only USB HID, but with the bluetooth HID devices also?  
With usage of `esp_hid` component?

Reply:
The bluetooth HID devices are not supported now in LVGL port. We can add it in future. 
Now, I propose leave the tasks in LVGL port like it is and change it in future, when USB HID component will be improved for better handling tasks. 
@igrr @tore-espressif What do you think?

Reply:
@espzav @roma-jam There are many new questions that rose from this usecase. Many of them require breaking changes and will take some time.

I'm OK with releasing it as-is for now, so we can demonstrate the feature to our users and optimize in follow-up MRs

Reply:
@roma-jam Please, when is the update of the USB HID planned?